app/helpers/application_helper.rb

```diff
@@ -364,4 +364,17 @@ def get_user_access_token(user)
   def pluralize_without_count(count, noun, text=nil)
     count.to_i == 1 ? "#{noun}#{text}" : "#{noun.pluralize}#{text}"
   end
+
+  # get the rails route specification in a format suitable
+  # for use in analytics or logging
+  # e.g. /single_cell/study/:accession/:study_name
+  def get_route_spec_string
+    spec_name = 'undefined'
+    Rails.application.routes.router.recognize(request) do |route, match|
+      spec_name = route.path.spec.to_s
+      # remove trailing term that gets auto-added by actionDispatch
+      spec_name = spec_name.sub('(.:format)', '')
```

Comment:
For each declaration in `routes.rb`, e.g. in:

```
get 'studies/:accession/submissions/:submission_id', to: 'site#get_study_submission', as: :site_get_study_submission
```

`spec` would be the part at the end, e.g. `site_get_study_submission`, right?  If so, could you replace underscores with hyphens (`site-get-study-submission`), to help consistent Mixpanel UX for Terra and SCP?

Also along those lines, could you ensure that the analytics page name for the home page is `root`, [as in Terra](https://broadinstitute.slack.com/archives/C015QCR0SKY/p1595353169002300)?

Reply:
yeah, I can do both those